### ISSUE TITLE
Nvidia Temp / Clock / Memory Stats

### ISSUE THREAD

**Issue by luisalvarado** [corecoding/Vitals#313]
### Has this issue been covered in the Wiki?

- [X] I have read the Wiki and my issue is not covered.

### Is there an existing issue reported already?

- [X] I have searched the existing issues, it does not exist.

### Describe the new feature you would like

This is a GREAT TOOL to say the least. But I have been using it together with Freon which is also another great tool. The issue is, I use both because Freon offers the overall CPU temp and the Nvidia card temp as seen here:

![image](https://user-images.githubusercontent.com/29236/213925989-1cb3abaa-c138-4b7b-b293-032569a1d875.png)

This is how it looks right now:

![image](https://user-images.githubusercontent.com/29236/213926000-8e508af2-51f5-4cf1-827b-0e62622350ab.png)

So a great option would be to detect a video card temp and maybe rename the average CPU temp option because right now it shows as Processor 0 which confused me a bit for a 13900k which has 32 threads.

![image](https://user-images.githubusercontent.com/29236/213926078-df131c01-d357-46d5-b549-7bdb0388c64f.png)

**Comment by w-flo** [comment]
The wiki says "It comes down to whether sensor information [for the GPU] is available in /proc or /sys", and as far as I know, the proprietary nvidia driver allows querying the temperature sensor through the nvidia-smi command only, it's not in /proc or /sys.

The wiki also says "Vitals uses asynchronous file reading to help prevent your system from lagging on every refresh. Executing a command line utility would cause unresponsiveness and is not something I want Vitals to exhibit." But I think it might be possible to prevent this unresponsiveness by calling the nvidia-smi tool through the async gio SubProcess or SubProcessLauncher.

I might have some time to give that a try and submit a pull request if I get it to work, if that would be acceptable for Vitals maintainers?

**Comment by corecoding** [comment]
@w-flo Thank you for quoting the Wiki. I get this request one or two times per month. At this point, I am definitely open to a pull request.

My previous reason for saying no was based around system freezing bug reports, as in my opinion shelling to the disk can sometimes cause this. Users of Vitals are very sensitive to this, but if we add a checkbox in settings (that part I can help with) then that allows the users to decide if they want to poll their video card stats.

**Comment by luisalvarado** [comment]
Thanks for the feedback. The idea of adding this is good, but after reading your comments and all against it now. Super bad idea if it needs to query nvidia-smi.

Maybe an idea would be to query in bigger intervals the GPU related information, instead of more realtime like the rest of the data for Vital. Would this also mean Freon is lagging the system or creating too many calls?

**Comment by w-flo** [comment]
I attempted to make nvidia-smi queries work without launching a new subprocess every couple of seconds, and it seems to work for me in preliminary testing. See #315 

Maybe there is some issue with this approach that I'm missing though.

**Comment by corecoding** [comment]
I will review this in detail tonight. The technique described sounds like a perfect compromise!

**Comment by corecoding** [comment]
I have merged @w-flo's work into the develop branch. I was unable to test myself because my Linux laptop is using Nouveau and I don't see an easy way to switch it to nvidia drivers under Fedora.

@luisalvarado can you test the develop branch and share your findings?

**Comment by luisalvarado** [comment]
Yes sir. Will test today. Thank you.

**Comment by luisalvarado** [comment]
Here are my findings:

TEMPERATURE:

![image](https://user-images.githubusercontent.com/29236/216784415-86b17c90-52c8-4386-ba54-c426b7734bda.png)

FAN SPEED:

![image](https://user-images.githubusercontent.com/29236/216784452-93518a79-ebaa-4564-9515-7661871e2897.png)

FINAL LOOK:

![image](https://user-images.githubusercontent.com/29236/216784799-f7e983ac-1d50-4be4-a954-53ac2c9798f0.png)


The ones I could see that could be worth it are:

CURRENT POWER USAGE
CURRENT MEMORY USAGE
CURRENT MEMORY SPEED
CURRENT CLOCK SPEED
TOTAL GPU UTILIZATION

If it helps, the line that I use to get the information I normally gather from the video card is:

nvidia-smi --query-gpu=timestamp,name,pci.bus_id,driver_version,pstate,pcie.link.gen.max,pcie.link.gen.current,temperature.gpu,utilization.gpu,utilization.memory,memory.total,memory.free,memory.used,power.draw,clocks.sm,clocks.mem,clocks.gr --format=csv -l 1

And that shows the following (Look at the last options which are the most useful):

![image](https://user-images.githubusercontent.com/29236/216784790-803c4a42-2a54-4463-b9e1-fcef76ca0ed5.png)

Did this help?

**Comment by w-flo** [comment]
Thanks for testing, @luisalvarado! So this is the command I currently use for the Vitals implementation because fan speed and temperature (and voltage, but that's not available in nvidia-smi) are "general purpose" sensor groups in Vitals:
`nvidia-smi --query-gpu=name,temperature.gpu,fan.speed --format=csv,noheader,nounits -l [seconds]`.

I agree that the other sensors, like power usage / GPU memory usage / clock speeds / GPU utilization would be interesting. But we would probably need a new "GPU" main level group in Vitals, next to "Temperature", "Fan", "Memory" and so on, because these sensors don't really fit into any existing group, I believe. For example, the existing "Memory" group focuses on system main memory, so I'm not sure if VRAM could be added to the same group. So the UI could be pretty confusing in the end (I'm afraid I'm a terrible UI designer).


About the fan speed not updating properly and being grayed out in the one screenshot: That means there was an error, maybe a bug in my code? For example, it will happen if you kill the nvidia-smi subprocess spawned by Vitals or if the nvidia-smi process exits for other reasons. Like when reading the process output fails. I haven't implemented an "auto-restart" feature because I was worried about spamming the system with repeated attempts to re-start nvidia-smi, and I guess the process should just stay alive if everything is working correctly.

If this happens without you manually killing the nvidia-smi process, I need to look into it. Was there anything "special" that you did before you noticed this issue, like maybe suspending the system for some time, or changing some nvidia driver settings, or anything that I could do to reproduce the issue? It has never happened to me (only when I "killall nvidia-smi" on purpose for testing). You should be able to restart the background process if this happens again by changing the update interval in Vitals settings, or disabling/re-enabling the experimental Nvidia feature in Vitals settings.

**Comment by luisalvarado** [comment]
Oh my, my apologies, the Fan was not grayed out. It was working, I happen to have the mouse over the option when I took the picture, but at that moment it was showing proper value. It is after (a couple of seconds later, when it stopped getting proper value). Sorry for the confusion.

In regards to suspending or anything, no. Just started taking pictures and showing you the results. Actually while I type it shows like this:

![image](https://user-images.githubusercontent.com/29236/216792316-fb474a6a-432c-44e8-bded-81ac63a94b93.png)

For restarting, actually yes, if I reboot it works again for a couple of seconds. Are you by chance the UI Designer of the app?

**Comment by w-flo** [comment]
I'm not the UI designer, this piece of code that fetches data from nvidia-smi is the first time I've worked with this app's code.

Alright, it's surprising that it just stays at 0%! I will look into this next week. Another issue is that "min" and "avg" fan speeds are now calculated incorrectly, for example 25% is counted as "0.25 RPM", and I'm not really sure what to do about it.

Also, I'm worried about memory use. The gnome-extensions "gjs" process is at 83,4 MB for me, according to gnome-system-monitor, and I don't remember it taking up significant amounts of memory. Maybe that's normal for Vitals though, I had only used it for one day before starting the nvidia-smi related development. But I do wonder if I've introduced some kind of memory leak somewhere, maybe only after using it for a longer time.

**Comment by luisalvarado** [comment]
Normally on the video cards (Well Nvidia in this case), it does not tell you the actual RPM but the percent in which it is supposed to be running. Fan RPM will vary by model and even inside the same card (Like here I have 3 fans but 2 of them run a bit faster than the last one). So I would stick with the percent and not worry about the RPM.

For the gjs, it shows 16MB on my end, but I just started using the computer like an hour ago.

**Comment by w-flo** [comment]
Well, maybe the way fan speed average / minimum / maximum values are calculated should be changed when percentage values are involved, at least. Maybe just ignore percentages in those calculations. Let's see what @corecoding thinks about this :-)

Another possible issue: Recently, gnome-shell likes to crash just before my system is going into suspend-to-RAM mode (literally in the last second before suspending): `kernel: gnome-shell[1480]: segfault at 20 ip 00007fc845f3c990 sp 00007fff49b8c9a0 error 4 in libmutter-11.so.0.0.0[7fc845e4a000+149000]`. That's annoying because all applications will crash because of a broken pipe once suspend-to-RAM ends (on wayland) and I have to re-login. I'm not sure if that could be related to my changes in Vitals? I don't think this happened before, but now it happened 3 times in 2 days. Maybe something is not handling "async read in gjs is active while going into suspend" so well and it somehow affects that gnome-shell process, even though Vitals should run in the gjs process? Or maybe because I'm actively working on the extension files while it is loaded and enabled.

Anyway, I'll try to see if I can reproduce the "fan speed doesn't update after it hit 0" issue today.

**Comment by w-flo** [comment]
I just sent a pull request #317 that might fix the issue with nvidia-smi values sometimes not updating. It was pretty rare in my testing and happened only when changing the update interval, so maybe there is some other condition that could trigger this (maybe related to changing settings in the nvidia-driver?).

**Comment by luisalvarado** [comment]
Thank you for the update @w-flo

That was a lot of testing.

**Comment by w-flo** [comment]
Thank you @luisalvarado for noticing that "fails to update" bug! :-) We really tested a lot.

The "gjs uses more memory" thing seems to be fine. Apparently, the extension code actually runs in the gnome-shell process itself, not the gjs "gnome.Shell.Extensions" process. I'm not 100% sure what the gjs process does, maybe it runs settings windows. Anyway, I'm pretty sure I haven't noticed that process before because I didn't interact with extension settings etc. So that should be okay.

So I believe the only remaining known issues are:
- What to do about percentage fan speed? Just ignore percentages while calculating the average, min and max values?
- Should we add more GPU data like GPU utilization, memory data, and so on – and what's a good UI design for that? Maybe a new sub-menu (group) for each Nvidia GPU installed on the system could work.

**Comment by luisalvarado** [comment]
Awesome. Well for the questions here you go

What to do about percentage fan speed? Just ignore percentages while calculating the average, min and max values?

Actually show only the percentage and ignore any other values. For fans it is rare and also difficult /not useful to measure the rpm for a video card.


Should we add more GPU data like GPU utilization, memory data, and so on – and what's a good UI design for that? Maybe a new sub-menu (group) for each Nvidia GPU installed on the system could work

Give me a moment for this one. I can hire a designer for it and provide an idea for it with her.

**Comment by corecoding** [comment]
Is there anyone else who has an Nvidia card who can test this dev build?

**Comment by luisalvarado** [comment]
@corecoding Hi, I tested with an Nvidia 4090 and 1080 last time. Should I test again?

**Comment by corecoding** [comment]
@luisalvarado I haven't made any changes to this feature, I was hoping someone else from the community could give it a try... May take a moment for someone to chime in though, as users have to proactively go to this page

**Comment by luisalvarado** [comment]
You mean I can go in and and submit my pull request?

**Comment by chrismou** [comment]
@corecoding I have a laptop running a 1050ti and a (new) laptop running a 3070ti, both using the Nvidia driver, I'm happy to test it on both but I'm afraid Ive not done any Gnome extension development before so am fairly new to this. 

If you can give me some hints on how to get the dev branch running and what info you need back from me, I'll spin it up on both this week and see what comes back

**Comment by chrismou** [comment]
@corecoding 

1050ti
![image](https://user-images.githubusercontent.com/319883/222141158-7e652f0f-d8e9-4e4f-95a6-149ec2413fd8.png)

3070ti

Tested on the 3070ti by running the OpenGl benchmarking test `__NV_PRIME_RENDER_OFFLOAD=1 __GLX_VENDOR_LIBRARY_NAME=nvidia glmark2` which showed a jump of 16c within a few seconds, so appears to be updating correctly

Only thing I'd say, and this is probably outside the scope of this, is that it might be nice to have different icons for different temperature types. I imagine that if you wanted temperatures visible on the top bar, CPU and GPU temp would be a common combination, but with them both using the same thermometer icon it's not immediately clear which is which. 

Is there anything else you want me to test?

**Comment by luisalvarado** [comment]
I was talking to the visual designer I work with, she did some of the icons like this:

![image](https://user-images.githubusercontent.com/29236/222321260-0222680e-8132-4579-8f25-9e7025c5cb35.png)

![image](https://user-images.githubusercontent.com/29236/222321285-02e1a940-e801-49e3-adf5-3ecb4b226d56.png)

These would be the base for the GPU section and then a smaller svg next to it showing the Temp,  Fan Speed, CPU Clock. The sub icons would be tied to the GPUs icons here so anyone looking at them knows "A ok this is the GPU Temp" and not confuse it with the CPU Temp. Same for clock, fan speed, etc..

**Comment by corecoding** [comment]
Moved the code to the https://github.com/corecoding/Vitals/tree/nvidia branch, I released support for gnome 44 today and needed to use that branch because it had other changes.

**Comment by luisalvarado** [comment]
Hey no worries @corecoding thank you for that. Do note, if I update vital and I am using Xorg, it will crash. 
![image](https://user-images.githubusercontent.com/29236/224551825-56a95dad-6f10-4713-b599-4666cc25abac.png)

I am guessing this is still been tested. i will check out the nvidia branch. Thank you.

**Comment by corecoding** [comment]
> Hey no worries @corecoding thank you for that. Do note, if I update vital and I am using Xorg, it will crash. ![image](https://user-images.githubusercontent.com/29236/224551825-56a95dad-6f10-4713-b599-4666cc25abac.png)
> 
> I am guessing this is still been tested. i will check out the nvidia branch. Thank you.

https://github.com/corecoding/Vitals/wiki/Broken-after-update should help.

**Comment by btotta** [comment]
![Captura de tela de 2023-03-13 17-15-12](https://user-images.githubusercontent.com/55134235/224822576-c72d3881-8583-4c16-9c03-9cc3e4bb611e.png)
![Captura de tela de 2023-03-13 17-16-56](https://user-images.githubusercontent.com/55134235/224822581-e6bda0c6-97cc-4f16-9225-537051bf7d32.png)

Idk if tests are still needed, but tried now with the nvidia branch and looks good, immediately update with running a game.

**Comment by corecoding** [comment]
@luisalvarado @w-flo I pushed some changes to the `nvidia` branch. I think the best way to handle GPU metrics is by adding a dedicated GPU section. This is rough draft code, but I wanted to share it to encourage contribution. I will make the menu dropdown GPU instead of Gpu. 

Can you see if my changes are working? I'd love to see a screenshot with the GPU sensor values being in the GPU section.

**Comment by trofosila** [comment]
All data displayed seems to be correct. I'm running a _strange_ fanless RTX A2000, the temp on it it's 37 °C, "Fan 0 rpm" is also correct as at the time of the screenshot the CPU fan was off (I'm running a semi-passive PC with a single fan that kick-in when the CPU temp is above 50 °C). Don't know what "Gpu: no data" means.

![image](https://user-images.githubusercontent.com/14992222/233184645-8703a0a1-6f00-4308-9b2d-128539850825.png)

**Comment by chrismou** [comment]
I'm not sure whats happened since the last update but I swtched to the `nvidia` branch and now the GPU has disappeared from the temperature menu, and I have a new GPU menu showing "No Data".

![image](https://user-images.githubusercontent.com/319883/233812513-ceb1c804-b304-4fb0-8729-11d0f30e1133.png)

This was working on the old branch https://github.com/corecoding/Vitals/issues/313#issuecomment-1450180725

**Comment by luisalvarado** [comment]
Sorry for being so absent. Was a beta tester for Ubuntu 23.04. Okay so Ubuntu 23.04, What I am seeing is that the option for enabling Nvidia will automatically turn itself off even if you enable it:

![image](https://user-images.githubusercontent.com/29236/234305043-9bf6f3ca-b528-4958-b14f-e9f47485dd48.png)


If you enable it, even though it shows enabled, it is actually off. Because of this, it does not show the Nvidia option in the menu.

**Comment by luisalvarado** [comment]
I had to reboot (The option as mentioned above would auto disable itself), but afterwards it shows like this:

![image](https://user-images.githubusercontent.com/29236/235269672-557864bc-e84a-42d2-b488-ae9b93f578a7.png)

**Comment by xulres** [comment]
The easiest way is to check if nvidia-smi provides the desired info if you call it directly. I would also suggest adding TOTAL GPU USAGE just to have parity with the CPU Options.

**Comment by trofosila** [comment]
@corecoding any plans to merge this to main?

**Comment by corecoding** [comment]
@trofosila It is not code complete yet. Not sure if the others want to put more time into this feature?

**Comment by xulres** [comment]
I'll get on it tomorrow one question @corecoding I believe it's better to keep the Fan and the Temperature in their respective categories. I would add usage and power (wattage) under the GPU menu point.

**Comment by corecoding** [comment]
@xulres sorry, didn't know you were waiting on me! I think your approach sounds perfect. 

Btw, I did make a recent changed that moved the core Gnome files into a sub directory in the repo, if you don't want to deal with the (likely) code conflicts, I can clean things up.

**Comment by xulres** [comment]
Added an Icon (selfmade) and refactored everything gpu related into the gpu menu added some other stats like utilization, memory utilization and power draw.

**Comment by corecoding** [comment]
Making note of these two GPU related tickets, I am guessing they can be closed when this feature is released.

https://github.com/corecoding/Vitals/issues/366
https://github.com/corecoding/Vitals/issues/359

**Comment by RyzenDew** [comment]
any update on this?

**Comment by RyzenDew** [comment]
Guess thats dead also

**Comment by corecoding** [comment]
This is a community driven feature. I don't have the hardware to test it myself, plus (as mentioned above) using command line utilities to grab sensors can create lags which, at the core, is why I created Vitals. At the time, some other system monitoring tools created lag.

What I am looking for is to have a group of individuals (could be as many as 2 or 3) come in and say yes, this works, or no it doesn't and here is a patch to make it work. Once we get to a point where it works for everyone, I will merge it in.

**Comment by trofosila** [comment]
Hi @corecoding I have in the past tested it on Gnome 44 / Fedora 38 and everything was fine for me (see comment above).
I will retest the nvidia branch in the coming days on Debian 12 and Fedora 39. I'll report back. Hope others can confirm it works or report if there are any issues.

**Comment by chrismou** [comment]
Ive tested this previously. On March 1st last year it worked great with no visible lag.

But then after you said you'd moved the branch, I updated and tested again (April 23rd) and it had stopped working, though it wasnt clear why. Both attempts I posted comments on this issue.

I'm not sure if anythings changed between April and now, but I'll reinstall and check again next week

**Comment by luisalvarado** [comment]
I need to do my computer from scratch. Too much garbage so that's why I have not reported yet.

**Comment by luisalvarado** [comment]
Just tested, I do not see it anywhere. IS there a special parameter or something I gotta do. It was working fine last year if that helps,.

**Comment by corecoding** [comment]
Can you try the [develop-nvidia](https://github.com/corecoding/Vitals/tree/develop-nvidia) branch? This is the code from Arastais. I tried it and it appeared to be the best yet. I had one issue with N/A values and he has fixed it.

**Comment by luisalvarado** [comment]
@corecoding Yes now it shows on the settings

![image](https://github.com/corecoding/Vitals/assets/29236/8f456e9e-1eec-41e2-9f49-f2bbba40ad7b)


But there is no option to select the Nvidia 4090 or anything Nvidia related. I am using the one from today.

![image](https://github.com/corecoding/Vitals/assets/29236/55fdbb80-8664-4de2-b739-4e7f8a6ee741)

**Comment by luisalvarado** [comment]
I need to go back on what I said before. After I enabled the option for Nvidia, I had to reboot (I am guessing login out and in, but in my case I needed to reboot. After which I got this:

![image](https://github.com/corecoding/Vitals/assets/29236/ece1d6dc-2df1-4198-b641-f1808a6c3af9)


But I also enabled this option I never noticed:

![image](https://github.com/corecoding/Vitals/assets/29236/769c49a6-ba70-416a-9582-590e50708bfc)

**Comment by chrismou** [comment]
@corecoding Is there a minimum gnome shell version for this branch?  Am on 42.9 and see this:

![image](https://github.com/corecoding/Vitals/assets/319883/fb12135d-15b9-45aa-8672-82c2a8ea1612)

**Comment by luisalvarado** [comment]
@corecoding Any chance of pushing it to the gnome extensions. I am only asking because there is a video I have been meaning to do about all the tools I use on my channel at https://www.youtube.com/@xtremelinux and CoreVitals has been asked a couple of times now. I want to show the options it has and how easy it is to use.

**Comment by corecoding** [comment]
I appreciate that @luisalvarado! Sorry for the delay. I will test it on my sons laptop and if it works, I'll get it released tonight or tomorrow.

**Comment by luisalvarado** [comment]
Do not worry. Also got 2 kids. I feel you my friend.

**Comment by corecoding** [comment]
Was able to get some testing in tonight. The Linux laptop I have access to has one Nvidia GPU. I ended up creating a bash script that simulated the output of nvidia-smi with two video cards. Confirming that Graphics (in the menu) changed to Graphics 1 and Graphics 2. Pretty cool.

I merged the develop branch into the main branch. You won't see the update come through normal channels, but this will put it in the hands of the Arch/Manjaro users. They love bleeding edge things and should be quick to let us know if they have an issue.

I would like to wait a few more days just to make sure - Vitals has a lot of users - would hate to break something, especially unrelated to the GPU goodness.

**Comment by corecoding** [comment]
No bug reports have come in. This has been submitted to Gnome for review. Fingers crossed all goes well for the masses.

**Comment by corecoding** [comment]
Alright folks. The submission was declined. This is the reasoning:

1. Please don&#x27;t use deprecated modules (line 3 `helpers/subprocess.js`):

   [EGO Review Guidelines: deprecated modules](https://gjs.guide/extensions/review-guidelines/review-guidelines.html#do-not-use-deprecated-modules)

2. Please don&#x27;t create global variable (line 7 `helpers/subprocess.js`).
   You can move that to the functions using it.

Please use the review page to follow up:

**Comment by corecoding** [comment]
I pushed some changes to the dev branch. If you all can check out that branch and do a git pull that would be helpful. I wasn't able to test it yet, but it should work.

**Comment by luisalvarado** [comment]
Okay here is the git dev version right now for me:

![image](https://github.com/corecoding/Vitals/assets/29236/7c14342a-8dda-4934-acb5-82e6537bceb1)

![image](https://github.com/corecoding/Vitals/assets/29236/e408ce38-e649-447c-bc83-de38aeff0d29)

Would be cool to have like a "Version something" somewhere though. something like:

![image](https://github.com/corecoding/Vitals/assets/29236/1da9cfe5-be86-43d1-a00d-89973de5ff9d)

**Comment by corecoding** [comment]
Thank you @luisalvarado. We are getting close. I pushed changes a bit ago to fix the issues, and also added Gnome 46 support.

**Comment by chrismou** [comment]
> @corecoding Is there a minimum gnome shell version for this branch? Am on 42.9 and see this:
> 
> ![image](https://private-user-images.githubusercontent.com/319883/302240491-fb12135d-15b9-45aa-8672-82c2a8ea1612.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDg4NjQwNjgsIm5iZiI6MTcwODg2Mzc2OCwicGF0aCI6Ii8zMTk4ODMvMzAyMjQwNDkxLWZiMTIxMzVkLTE1YjktNDVhYS04NjcyLTgyYzJhOGVhMTYxMi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMjI1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDIyNVQxMjIyNDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01OGIxZmQyMTllYmE1YTBiMTdlNzQ4NjQ0MDAxMDYxNTdiNGMwODI0NGVmN2I3MGU1OTM0YzYxZDIyMWNlMzJhJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.pB3nPR_pNY08NolepLIc6kPaKVPEY4xYMTUCY5Q2wIo)

Any thoughts on this @corecoding ?  Still on 42.9, Ubuntu 22.04 LTS and still showing as Outdated. I can't see a mention of supported Gnome Shell versions on the readme so unsure if the plugin has dropped support for this version or its broken

**Comment by luisalvarado** [comment]
> > @corecoding Is there a minimum gnome shell version for this branch? Am on 42.9 and see this:
> > ![image](https://private-user-images.githubusercontent.com/319883/302240491-fb12135d-15b9-45aa-8672-82c2a8ea1612.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDg4NjQwNjgsIm5iZiI6MTcwODg2Mzc2OCwicGF0aCI6Ii8zMTk4ODMvMzAyMjQwNDkxLWZiMTIxMzVkLTE1YjktNDVhYS04NjcyLTgyYzJhOGVhMTYxMi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMjI1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDIyNVQxMjIyNDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01OGIxZmQyMTllYmE1YTBiMTdlNzQ4NjQ0MDAxMDYxNTdiNGMwODI0NGVmN2I3MGU1OTM0YzYxZDIyMWNlMzJhJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.pB3nPR_pNY08NolepLIc6kPaKVPEY4xYMTUCY5Q2wIo)
> 
> Any thoughts on this @corecoding ? Still on 42.9, Ubuntu 22.04 LTS and still showing as Outdated. I can't see a mention of supported Gnome Shell versions on the readme so unsure if the plugin has dropped support for this version or its broken

The image did not load. For the support, I think everything got updated to the latest. At least for the JS files I looked into.

**Comment by corecoding** [comment]
> > @corecoding Is there a minimum gnome shell version for this branch? Am on 42.9 and see this:
> 
> > 
> 
> > ![image](https://private-user-images.githubusercontent.com/319883/302240491-fb12135d-15b9-45aa-8672-82c2a8ea1612.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDg4NjQwNjgsIm5iZiI6MTcwODg2Mzc2OCwicGF0aCI6Ii8zMTk4ODMvMzAyMjQwNDkxLWZiMTIxMzVkLTE1YjktNDVhYS04NjcyLTgyYzJhOGVhMTYxMi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMjI1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDIyNVQxMjIyNDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01OGIxZmQyMTllYmE1YTBiMTdlNzQ4NjQ0MDAxMDYxNTdiNGMwODI0NGVmN2I3MGU1OTM0YzYxZDIyMWNlMzJhJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.pB3nPR_pNY08NolepLIc6kPaKVPEY4xYMTUCY5Q2wIo)
> 
> 
> 
> Any thoughts on this @corecoding ?  Still on 42.9, Ubuntu 22.04 LTS and still showing as Outdated. I can't see a mention of supported Gnome Shell versions on the readme so unsure if the plugin has dropped support for this version or its broken

This branch/feature requires gnome 45 or 46. For those that are wondering, older versions of Vitals are still available. 

Reason for this is each version of gnome breaks extensions, and code gets very clunky and hard to test when supporting older versions.

**Comment by chrismou** [comment]
Pretty frustrating if Gnome aren't making an effort to maintain backwards compatibility. Fair enough not wanting to support back 5+ versions if that's the case.

Well the new Ubuntu LTS (24.04) is out this April, so I guess I'll check back in a few months.

**Comment by corecoding** [comment]
The latest version was submitted again. Fingers crossed!

**Comment by corecoding** [comment]
Ok, it's released! Were y'all able to upgrade with success?

**Comment by luisalvarado** [comment]
Sorry, family and computer things came up. Yes I see it working properly here:

![image](https://github.com/corecoding/Vitals/assets/29236/f1f179cc-2b9d-4876-882f-f3e42366382d)

**Comment by corecoding** [comment]
No worries @luisalvarado! Thank you for checking in. Thank you for everyone's contributions! Ideation, Testing, Coding, and everything in-between. As you probably know, I was adamant that I didn't want a background command, but you showed me that y'all really wanted it.

I'm going to close this ticket soon. If there are issues or improvements, we will open a new ticket.
